packages/connectkit/src/components/DaimoPay.tsx

```diff
@@ -224,7 +231,40 @@ export const DaimoPayProvider = ({
 
   const log = debugMode ? console.log : () => {};
 
-  const paymentInfo: PaymentInfo = getPaymentInfo(setOpen, log);
+  // PaymentInfo is a second, inner context object containing a DaimoPayOrder
+  // plus all associated status and callbacks. In order for useContext() and
+  // downstream hooks like useDaimoPayStatus() to work correctly, we must set
+  // set refresh context when payment status changes; done via setDaimoPayOrder.
+  const [daimoPayOrder, setDaimoPayOrder] = useState<DaimoPayOrder>();
+  const paymentInfo: PaymentInfo = getPaymentInfo({
+    daimoPayOrder,
+    setDaimoPayOrder,
+    setOpen,
+    log,
+  });
+
+  // When a payment is in progress, poll for status updates. Do this regardless
+  // of whether the modal is still being displayed for useDaimoPayStatus().
+  useEffect(() => {
+    // Order just updated...
+    if (daimoPayOrder?.mode !== DaimoPayOrderMode.HYDRATED) return;
+
+    const { sourceStatus, intentStatus } = daimoPayOrder;
+    let intervalMs = 0;
+    if (sourceStatus === DaimoPayOrderStatusSource.WAITING_PAYMENT) {
+      intervalMs = 2500; // additional, faster polling in WaitingOther
+    } else if (intentStatus === DaimoPayIntentStatus.PENDING) {
+      intervalMs = 300; // poll fast from (payment initiated) to (finished)
+    } else {
+      return;
+    }
+
+    log(`[PAY] polling in ${intervalMs}ms`);
+    setTimeout(
+      () => retryBackoff("refreshOrder", () => paymentInfo.refreshOrder()),
```

Comment:
once refreshOrder succeeds, it calls setDaimoPayOrder, kicking off another round of polling ^